backend/pvm.pm

```diff
@@ -0,0 +1,159 @@
+}
+
+sub do_start_vm {
+    my $self = shift;
+    $self->unlink_crash_file;
+    $self->start_lpar();
+    return {};
+}
+
+sub runcmd {
+    diag "running " . join(' ', @_);
+    return system(@_);
+}
+
+sub pvmctl {
+    my $self = shift @_;
+    my $vars = \%bmwqemu::vars;
+    my ($type, $action) = @_;
+    push(my @cmd, $self->{pvmctl}, $type, $action);
+    my $lpar = $vars->{LPAR};
+
+    if ($type =~ /lpar/) {
+        push(@cmd, "-i", "name=" . $lpar) if ($action =~ /power|restart|delete/);
+        if ($action =~ /create/) {
+            my ($type, $action, $cpu, $memory) = @_;
+            push(@cmd, qw{--proc-type shared --sharing-mode uncapped --type AIX/Linux});
+            push(@cmd, "--proc", $cpu, "--mem", $memory, "--name", $lpar);
+            push(@cmd, "--proc-unit", $cpu * 0.05);
+        }
+    }
+    elsif ($type =~ /scsi/) {
+        my ($type, $action, $kind, $file) = @_;
+        push(@cmd, "--type", $kind, "--stor-id", $file);
+        push(@cmd, "--lpar", "name=" . $lpar);
+        push(@cmd, "--vg", "name=rootvg") if ($type =~ /lv/);
+    }
+    elsif ($type =~ /lv/) {
+        my ($type, $action, $name, $size) = @_;
+        push(@cmd, "--name", $name, "--size", $size);
+    }
+    elsif ($type =~ /eth/) {
+        my ($type, $action, $vlan, $vswitch) = @_;
+        push(@cmd, "--pvid", $vlan, "--vswitch", $vswitch, "-p", "name=" . $lpar);
+    }
+    else {
+        die "Unrecognized command $type";
+    }
+    runcmd(@cmd);
+}
+sub attach_console {
+    my $vars    = \%bmwqemu::vars;
+    my $vncport = qx{sudo /usr/sbin/mkvtermutil --id $vars->{LPARID} --vnc --local --log serial0 2>/dev/null};
+    $vncport =~ /([0-9]+)/;
+    chomp($vncport);
+    $vars->{VNC} = $vncport;
+    diag "VNC is $vars->{VNC}";
+}
+
+sub start_lpar {
+    my $self = shift;
+    my $vars = \%bmwqemu::vars;
+    #general settiings
+    $vars->{LPAR} = "osauto" . $vars->{WORKER_ID};
+    $vars->{CPUS} ||= 1;
+    $vars->{MEM}  ||= "2048";
+    #disk settings
+    $vars->{NUMDISKS}  ||= 1;
+    $vars->{HDDSIZEGB} ||= 15;
+    #network settings
+    $vars->{NIC}     ||= "sea";
+    $vars->{NICVLAN} ||= 1;
+    $vars->{VSWITCH} ||= "ETHERNET0";
+    $vars->{VIOISO} = md5_hex(basename($vars->{ISO})) . ".iso";
+    my $iso = $vars->{VIOISO};
+    die "$iso exceeds 37 characters\n" if (length($iso) > 37);
+
+    if (!($vars->{ARCH} =~ /ppc64/)) {
+        die "arch $vars->{ARCH} is not allowed on pvm backend";
+    }
+    #create lpar
+    $self->pvmctl("lpar", "create", $vars->{CPUS}, $vars->{MEM});
+
+    my $id = substr(qx/pvmctl lpar list -d LogicalPartition.id --where LogicalPartition.name=$vars->{LPAR}/, 3);
+    chomp($id);
+    $vars->{LPARID} = $id;
+    bmwqemu::save_vars();
+
+    #unfortunately NovaLink can't take files longer than 38 chars. So we need to shorten ISO name here
+    #we copy isos from nfs mount on VIO side to VMLibrary
+    my $source_iso = $vars->{ISO};
+    $source_iso =~ s/var\/lib\/openqa\/share\/factory\///g;
```

Comment:
I would feel better if you used 'iso/' . basename($source_iso) - in the light of https://progress.opensuse.org/issues/10470 it sounds bad to hardcode openQA paths in os-autoinst.